packages/kbn-esql-utils/src/utils/query_parsing_helpers.ts

```diff
@@ -40,14 +40,27 @@ export function hasTransformationalCommand(esql?: string) {
 }
 
 export function getLimitFromESQLQuery(esql: string): number {
-  const limitCommands = esql.match(new RegExp(/LIMIT\s[0-9]+/, 'ig'));
-  if (!limitCommands) {
+  const { ast } = getAstAndSyntaxErrors(esql);
+  const limitCommands = ast.filter(({ name }) => name === 'limit');
+  if (!limitCommands || !limitCommands.length) {
+    return DEFAULT_ESQL_LIMIT;
+  }
+  const limits: number[] = [];
+
+  walk(ast, {
+    visitLiteral: (node) => {
+      if (!isNaN(Number(node.value))) {
+        limits.push(Number(node.value));
```

Comment:
Haven't had a chance to review yet, but one thing I want to check is if this is too loose.

E.g. does the `10` in `FROM a | EVAL 10` get detected as a limit?

Reply:
No why ? I am parsing only the limit commands, eval is filtered out